### DISCUSSION TITLE
Como convencer meu chefe a trabalhar com uma stack moderna de tecnologias?

### DISCUSSION THREAD

**Discussion by Rodrigo54** [frontendbr/forum#1405] [Discussão]
Olá! Sou desenvolvedor full stack, moro em São Luís do Maranhão e tenho muito conhecimento em angular2 e express e estou trabalhando numa empresa como desenvolvedor mobile com ionic.

A stack da empresa é MSSQL 2000 no banco de dados que não está normatizado e é distribuído junto com os aplicativos, Delphi nas aplicações desktop. As aplicações são distribuídas para funcionarem nessa stack.

Tudo ainda é feito de forma muito _"artesanal"_ sem muita automação e embora eu tenha conseguido criar aplicações mobile para trabalhar nessa stack sinto que poderíamos ser mais produtivos se a stack de trabalho fosse uma mais atualizada.

talvez um mongodb numa nuvem e com as aplicações desktop em electron.

já consegui convence-los da necessidade de usar um banco em nuvem mas não consigo demostrar as vantagens de usar um noSQL ao invés de usar SQL.

Eu trabalho numa equipe de 3 onde um deles é o chefe/dono e sou o único formado e existe uma considerável diferença de idade entre eu e eles, a empresa também tem mais de 15 anos e tem muitos clientes em todo o estado. Sinto que fui contratado para trazer inovação para a empresa e embora eu entregue bons resultados creio que falho nessa tarefa ao não conseguir convencer eles a ~~saírem dos anos 90~~ migrarem de tecnologia 

Alguém sabe o que posso fazer pra melhorar esse cenário?

**Comment by MrS4w** [comment]
Primeiramente você tem que se perguntar o seguinte: "A empresa é de tecnologia?" se a resposta for "sim", eu diria até que é um pouco mais fácil porquê você vai ter que convencê-los do porquê a tecnologia que você está propondo é melhor e o quê a empresa ganharia em benefício utilizando uma outra stack, por outro lado, se a resposta for "não" o dono do negócio só está preocupado em entregar valor pros seus clientes, independente da tecnologia, de toda forma, você tem que conseguir provar com números reais de que a tecnologia que você está propondo pra eles é de fato boa, usada no mercado e afins.

**Comment by evertonthepaula** [comment]
Antes de tentar os convencer a deixar de usar SQl tente nos convencer do POR QUE você deveria deixar de usar SQL. Como você faria a migração de todo o sistema atual para trabalhar com serviços e APIs modernas desacoplando os clientes da regra de negócio? crie uma nova arquitetura para o sistema se for preciso, e será :) .

Se você conseguir convencer as pessoas aqui no fórum, vai ter argumentos para falar com seu chefe. 

Uma dica, apresente potenciais de crescimento para seu chefe, gastar horrores para modernizar seu sistema e continuar ganhando o mesmo, não me parece muito atrativo. Mostre para ele que olhar somente o presente e sentar em cima do problema vai matar se produto no futuro, se este for o caso obviamente, os novos concorrentes provavelmente já nascem com aplicações modernas que conseguem escalar para todos os lados, hoje ele tem desktop e mobile, com inúmeras "gambiarras" para tudo funcionar, e amanhã qual vai sera  necessidade? Se o ecossistema for escalável o amanhã não é problema, é só uma questão de implementação.

**Comment by helderberto** [comment]
Olá, @Rodrigo54. Tudo bem?

Acredito que um ponto importante, inclusive mencionado aqui mesmo é de fato as vantagens em relação a produtividade da equipe, custo para manutenção e ser colocado na balança se de fato a tecnologia que esta propondo é adequada ao cenário.

Alguns pontos que você pode utilizar para levar a sua equipe, são os seguintes:
* Qual o custo/benefício de trocar a tecnologia?
* Quantos pessoas da equipe estão aptas a esta mudança?
* No caso de mudanças na equipe, qual a dificuldade para encontrar novos profissionais?

Existem diversos outros tópicos que podem ser levados em consideração, porém esses que mencionei podem te ajudar a refletir se é válida esta mudança.

Espero ter ajudado de alguma forma.

Abraços!

**Comment by felipefialho** [comment]
Primeiro pergunte para você mesmo a necessidade de trocar, existem motivos convincentes que podem ser facilmente respondidos?

Após isso, caso tenha que migrar projetos antigos para a nova stack, faça uma analise do tempo que essa mudança levaria, pensa que muitas horas sua e do seu time serão gastas nisso.

Por último, sugiro fazer um levantamento de custo de tecnologias mais antigas e comparar com o custo ao usar tecnologias mais novas.

Coisas como:

- desenvolvimento
- manutenção
- segurança

Haverá de fato uma diminuição de custos que vai ser sentida pela empresa / dono?

Digo tudo isso, porque se ele não vê vantagem em usar stacks atuais, será muito difícil de convence-lo do contrário, especialmente se essa decisão não for pautada em números.

Depois nos atualize.
